### PR TITLE
Databases/week2

### PR DESCRIPTION
**Pull request by STEP0HEN**

### CODE REVIEW

File: databases/week2/databases-week2-Homework/more_queries.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+-- Get all taks for 'Donald Duck' where status 'not started' 
+SELECT t.title
+FROM task t
+INNER JOIN user_task ut ON t.id = ut.task_id
+INNER JOIN status s ON t.status_id = s.id 
+WHERE ut.user_id = (
+    SELECT id 
+    FROM user
+    WHERE name LIKE '%Donald Duck%'
```
Comment (babakbandpey) [inline databases/week2/databases-week2-Homework/more_queries.sql]: What would happen if there are more than one user with the name LIKE Donald Duck?

File: databases/week2/databases-week2-Homework/more_queries.sql [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+)
+AND s.id = (
+    SELECT id
+    FROM status
+    WHERE name = 'Not started'
+);
+
+
```
Comment (babakbandpey) [inline databases/week2/databases-week2-Homework/more_queries.sql:19]: This is a complicated way to bind queries.
I recommend to simplify it

File: databases/week2/databases-week2-Homework/school_databse.sql [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+`begins` DATETIME NULL DEFAULT NULL,
+`ends` DATETIME NULL DEFAULT NULL
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE `student`(
+`id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+`name` varchar(255) NOT NULL,
+`email` varchar(255) NOT NULL,
+`phone` varchar(255) NULL,
+`class_id` int(10) unsigned NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/databases-week2-Homework/school_databse.sql]: class_id is not needed when you have the student_classes table.

File: databases/week2/databases-week2-Homework/school_databse.sql [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+
+-- create index on name column from student table
+CREATE INDEX student_name_index
+ON student (name);
+
+-- Add column to table
+
+ALTER TABLE class ADD COLUMN status ENUM('not started', 'ongoing', 'finished')
```
Comment (babakbandpey) [inline databases/week2/databases-week2-Homework/school_databse.sql]: What would happen to rows that are already in the table when you add a new ENUM type field?

**Review by babakbandpey — CHANGES_REQUESTED** [review]
Good job, there are few questions to think about and change the SQL accordingly.
Looking forward to the fixes.

File: databases/week2/databases-week2-Homework/school_databse.sql (hunk 3 above)
Comment (STEP0HEN) [reply databases/week2/databases-week2-Homework/school_databse.sql]: okay

File: databases/week2/databases-week2-Homework/more_queries.sql (hunk 2 above)
Comment (STEP0HEN) [reply databases/week2/databases-week2-Homework/more_queries.sql:19]: okay

File: databases/week2/databases-week2-Homework/more_queries.sql (hunk 1 above)
Comment (STEP0HEN) [reply databases/week2/databases-week2-Homework/more_queries.sql]: I think If they pass the condition status.name = not started then they output as well if not, nothing will happen.

File: databases/week2/databases-week2-Homework/school_databse.sql (hunk 4 above)
Comment (STEP0HEN) [reply databases/week2/databases-week2-Homework/school_databse.sql]: I think they will not be affected until we specifically alter them as well with some conditions

**Comment by STEP0HEN** [comment]
Thanks again for the review @babakbandpey , I implemented the changes.

File: databases/week2/databases-week2-Homework/school_databse.sql (hunk 4 above)
Comment (babakbandpey) [reply databases/week2/databases-week2-Homework/school_databse.sql]: The problem here will become obvious when the table has records and you add new column and the existing rows will be missing data in that column. Add a Default value to it.

**Review by babakbandpey — CHANGES_REQUESTED** [review]
One last thing and you will be done.
Good work Stephen.

**Comment by STEP0HEN** [comment]
@babakbandpey  thanks for the reviews, I modified the status column as you suggested and added a default value.

**Review by babakbandpey — APPROVED** [review]
Thank you for the good work. Keep it up.

**Pull request merged**